src/material/dialog/dialog-container.ts

```diff
@@ -117,11 +134,11 @@ export class MatDialogContainer extends _MatDialogContainerBase implements OnDes
   private _hostElement: HTMLElement = this._elementRef.nativeElement;
   /** Duration of the dialog open animation. */
   private _openAnimationDuration = this._animationsEnabled
-    ? numbers.DIALOG_ANIMATION_OPEN_TIME_MS
+    ? parseCssTime(this._config.enterAnimationDuration) ?? numbers.DIALOG_ANIMATION_OPEN_TIME_MS
```

Comment:
Couldn't we just switch it to use Angular animations? Since MDC is exposed the animation variables already, we should be able to use them in the animation definition too.

Reply:
Hmm, possibly. I think I could also just deprecate the string duration parameters and add new ones where we ask people to pass numbers. That way I could get rid of the time parsing logic and make this pretty trivial, even with just plain CSS